Add breaking change doc for isFocusable (#12159)
flutter/website: src/content/release/breaking-changes/deprecate-focusable.md

---

> title: Deprecate  `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable`

Reviewer: ```suggestion
title: Deprecate `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable`
```

description: >
  The `focusable` parameter has been replaced by `isFocused`.
---

## Summary

> The `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable` parameters were deprecated
> in favor of the `SemanticsProperties.focused` and `SemanticsConfiguration.isFocused` parameter.

Reviewer: ```suggestion
The `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable`
parameters were deprecated in favor of the `SemanticsProperties.focused` and
`SemanticsConfiguration.isFocused` parameters.
```

## Context

> The `SemanticsConfiguration.isFocusable` is a boolean to describe if the semantics node can have input focus, and `SemanticsConfiguration.isFocused` is a boolean to describe if the semantics node has input focus.

Reviewer: ```suggestion
The `SemanticsConfiguration.isFocusable` is a boolean to describe
whether the semantics node can have input focus, and
`SemanticsConfiguration.isFocused` is a boolean to describe if the
semantics node has input focus.
```

> same with `SemanticsProperties.focusable` and `SemanticsProperties.focused`.

Reviewer: ```suggestion
The fix is the same with `SemanticsProperties.focusable` and `SemanticsProperties.focused`.
```

## Description of change

The `SemanticsConfiguration.isFocusable` is deprecated in

> favor of `SemanticsConfiguration.isFocused`, which is a nullable boolean, setting it to true/false will automatically make isFocusable to `true` and setting it to null will make isFocusable to false.

Reviewer: ```suggestion
favor of `SemanticsConfiguration.isFocused`, which is a nullable boolean;
setting it to true/false automatically sets `isFocusable` to `true`,
and setting it to null sets `isFocusable` to `false`.
```

Really?

## Migration guide

> Replace `SemanticsConfiguration.isFocusable` with `SemanticsConfiguration.isFocused`

Reviewer: ```suggestion
Replace `SemanticsConfiguration.isFocusable` with `SemanticsConfiguration.isFocused`.
```

> ### Example 1: setting isFocused to `true` or `false` will automatically set isFocusable to `true`.

Reviewer: ```suggestion
### Example 1: setting `isFocused` to `true` automatically sets `isFocusable` to `true`.
```
???

Author: set isFocused to a value that is not null, `true` or `false` -> set isFocused to `true` 
set isFocused to `null` -> set isFocused to `false`

Code before migration:

```dart
void describeSemanticsConfiguration(SemanticsConfiguration config) {
  config.isFocusable = true;
  config.isFocused = true;
}
```

Code after migration:

```dart
void describeSemanticsConfiguration(SemanticsConfiguration config) {
  config.isFocused = true;
}
```

> ### Example 2: setting isFocused to null will automatically set isFocusable to `false`.

Reviewer: ```suggestion
### Example 2: setting `isFocused` to null automatically sets isFocusable to `false`.
```

Code before migration:

```dart
void describeSemanticsConfiguration(SemanticsConfiguration config) {
  config.isFocusable = false;
  config.isFocused = false;
}
```

Code after migration:

```dart
void describeSemanticsConfiguration(SemanticsConfiguration config) {
  config.isFocused = null;
}
```

## Timeline

Landed in version: Not yet<br>
In stable release: Not yet

> 

Reviewer: Is this in the main channel yet? If so, when did it land? Please add that info here.

Author: the PR is still pending review, will add the info after the PR is landed

---
title: Deprecate `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable`
description: >
  The `focusable` parameter has been replaced by `isFocused`.
---

## Summary

The `SemanticsProperties.focusable` and `SemanticsConfiguration.isFocusable`

> parameters were deprecated in favor of the `SemanticsProperties.focused` and

Author: done

`SemanticsConfiguration.isFocused` parameters.

## Context

The `SemanticsConfiguration.isFocusable` is a boolean to describe
whether the semantics node can have input focus, and
`SemanticsConfiguration.isFocused` is a boolean to describe if the
semantics node has input focus.

> The fix is the same with `SemanticsProperties.focusable` and `SemanticsProperties.focused`.

Reviewer 2: This section should talk about why this we decide to deprecate isFocusable

Author: done
